src/main/java/com/denizenscript/depenizen/bukkit/objects/luckperms/LuckPermsTrackTag.java

```diff
@@ -130,43 +139,83 @@ public ObjectTag getObjectAttribute(Attribute attribute) {
         // @description
         // Returns the name of the track.
         // -->
-        if (attribute.startsWith("name")) {
-            return new ElementTag(track.getName()).getObjectAttribute(attribute.fulfill(1));
-        }
+        tagProcessor.registerTag(ElementTag.class, "name", (attribute, object) -> {
+            return new ElementTag(object.getTrack().getName());
+        });
 
         // <--[tag]
         // @attribute <LuckPermsTrackTag.groups[(<player>)]>
         // @returns ListTag
         // @plugin Depenizen, LuckPerms
         // @description
-        // Returns the list of groups in the track.
+        // This returns names of the groups instead of <@link objecttype LuckPermsGroupTag>s, as groups can be unloaded from LuckPerms but still be part of the track.
+        // If a player input is specified, limits to only the groups that the player is in.
+        // -->
+        tagProcessor.registerTag(ListTag.class,"groups", (attribute, object) -> {
+            List<String> trackGroups = object.getTrack().getGroups();
+            if (!attribute.hasParam()) {
+                return new ListTag(trackGroups);
+            }
+            PlayerTag player = attribute.paramAsType(PlayerTag.class);
+            if (player == null) {
+                attribute.echoError("Invalid player input for 'group' tag.");
+                return null;
+            }
+            User user = LuckPermsBridge.luckPermsInstance.getUserManager().getUser(player.getUUID());
+            if (user == null) {
+                return null;
+            }
+            List<String> memberGroups = user.resolveInheritedNodes(QueryOptions.nonContextual()).stream()
+                    .filter(NodeType.INHERITANCE::matches)
+                    .map(NodeType.INHERITANCE::cast)
+                    .filter(net.luckperms.api.node.Node::getValue)
+                    .map(InheritanceNode::getGroupName)
+                    .filter(trackGroups::contains).toList();
+            return new ListTag(memberGroups);
```

Comment:
This (and the other list return in this tag) should probably be plain-text as well